lib/state_machine/state_machine.cpp

```diff
@@ -23,4 +23,14 @@ State StateMachine::getCurrentState()
   return current_state_;
 }
 
+core::Result StateMachine::handleMessage(const Message &message)
+{
+  const auto temp = transition_to_state_.find({current_state_, message});
```

Comment:
`next_state` maybe? at least something more descriptive than `temp`

(nice work on the `const` btw)